### ISSUE TITLE
[expo-notifications] NotificationContentInput with sound set to true or string "default" didn't play sound on IOS only

### ISSUE THREAD

**Issue by guanhuay** [expo/expo#7659]
## 🐛 Bug Report

I just tried integrate the newly released SDK 37 expo-notifications into my project. 

I discover one bug with it. It is while calling the scheduleNotificationAsync() with the following payload, when set sound property to boolean true or string "default". On IOS it plays no sound.

### Environment

  Expo CLI 3.17.11 environment info:
    System:
      OS: macOS Mojave 10.14.6
      Shell: 3.2.57 - /bin/bash
    Binaries:
      Node: 12.12.0 - /usr/local/bin/node
      Yarn: 1.21.1 - /usr/local/bin/yarn
      npm: 6.11.3 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.5 AI-191.8026.42.35.6010548
      Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 3.17.11

Running on bare workflow + expo unimodules
iOS target 9
Android target: min 21

### Steps to Reproduce

1. Set up a minimal bare workflow project with expo
2. Add "expo-notifications" as dependencies
3. Use the following code snippet to schedule a notification

```
import React from "react";
import { Button, StyleSheet, Text, View } from "react-native";
import * as Notifications from "expo-notifications";

async function requestPermissionsAsync() {
  return await Notifications.requestPermissionsAsync({
    ios: {
      allowAlert: true,
      allowBadge: true,
      allowSound: true,
      allowAnnouncements: true,
    },
  });
}

Notifications.setNotificationHandler({
  handleNotification: async (notification) => {
    return {
      shouldShowAlert: true,
      shouldPlaySound: true,
      shouldSetBadge: false,
    };
  },
});

export default function App() {
  React.useEffect(() => {
    requestPermissionsAsync();
  }, []);

  return (
    <View style={styles.container}>
      <Button
        title="Show notification"
        onPress={() => {
          Notifications.scheduleNotificationAsync({
            content: {
              title: "A title",
              body: "A body",
              sound: true // or sound: "default"
            },
            trigger: {
              seconds: 1,
            },
          });
        }}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
```

With this code snippet

### Expected Behavior

On IOS, it should play sound

### Actual Behavior

On IOS, no sound was played on notification shows up

### Reproducible Demo

Run this repo on IOS ===> [From @brentvatne ](https://github.com/brentvatne/local-notif-ios-example)

**Comment by sjchmiela** [comment]
Is it possible that if you had changed the handler function to return `shouldPlaySound: true` in the object, the sound would be played?

```diff
 Notifications.setNotificationHandler({
   handleNotification: async (notification) => {
     return {
       shouldShowAlert: true,
-      shouldPlaySound: false,
+      shouldPlaySound: true,
       shouldSetBadge: false,
     };
   },
 });
```

**Comment by guanhuay** [comment]
> Is it possible that if you had changed the handler function to return `shouldPlaySound: true` in the object, the sound would be played?
> 
> ```diff
>  Notifications.setNotificationHandler({
>    handleNotification: async (notification) => {
>      return {
>        shouldShowAlert: true,
> -      shouldPlaySound: false,
> +      shouldPlaySound: true,
>        shouldSetBadge: false,
>      };
>    },
>  });
> ```

@sjchmiela Sorry about that typo up there since I just copy paste the example from @brentvatne. 

But i just changed showPlaySound to true, and it still doesn't play sound, it is very easily to reproduce.

**Comment by sjchmiela** [comment]
Hm, I have reproduced the issue successfully, and it turns out that the sound does play, but only if the notification is shown when the app is not in foreground. I couldn't have made the phone make a sound while showing a notification of the foreground app.

If you schedule a notification for a moment when the app will be in foreground, does the phone play a sound (or vibrate, if in silent mode)?

**Comment by guanhuay** [comment]
> Hm, I have reproduced the issue successfully, and it turns out that the sound does play, but only if the notification is shown when the app is not in foreground. I couldn't have made the phone make a sound while showing a notification of the foreground app.
> 
> If you schedule a notification for a moment when the app will be in foreground, does the phone play a sound (or vibrate, if in silent mode)?

If I understand correctly, you want me to try running app in foreground, then show a notification?

Yes I already tried it, no sound was played from the device, is this expected or?

**Comment by sjchmiela** [comment]
I also am not sure I understand you correctly, so let me rephrase my proposal with more code and a step-by-step instructions: 🙂 

1. Call
    ```ts
    Notifications.scheduleNotificationAsync({
      content: {
        title: "I made a sound",
        sound: true
      },
      trigger: { seconds: 5 }
    });
    ```
2. Press the home button, effectively backgrounding the app.
3. After 5 seconds the notification should be displayed (the app is still in background).
4. The notification should have played a sound (if the device is not in silent mode) or made the phone vibrate (if the device is in silent mode). Unfortunately (fortunately for the users 😃) it is not possible to make the notification play a sound when the phone is in silent mode.

What effect does this following these instructions have?

**Comment by guanhuay** [comment]
> I also am not sure I understand you correctly, so let me rephrase my proposal with more code and a step-by-step instructions: 🙂
> 
> 1. Call
>    ```ts
>    Notifications.scheduleNotificationAsync({
>      content: {
>        title: "I made a sound",
>        sound: true
>      },
>      trigger: { seconds: 5 }
>    });
>    ```
> 2. Press the home button, effectively backgrounding the app.
> 3. After 5 seconds the notification should be displayed (the app is still in background).
> 4. The notification should have played a sound (if the device is not in silent mode) or made the phone vibrate (if the device is in silent mode). Unfortunately (fortunately for the users 😃) it is not possible to make the notification play a sound when the phone is in silent mode.
> 
> What effect does this following these instructions have?

By following your instruction, sound was played, worked like a charm.
Maybe I was doing something wrong.
